### PR TITLE
Climb roll2

### PR DESCRIPTION
**Pull request by AAgraviador3176**
merge and integrate climbroll

### CODE REVIEW

File: src/main/java/team3176/robot/subsystems/superstructure/climb/Climb.java [hunk 1]
```diff
@@ -117,24 +148,17 @@ public Command moveLeftRightPosition(DoubleSupplier deltaLeft, DoubleSupplier de
...
     io.updateInputs(inputs);
     Logger.processInputs("Climb", inputs);
     pid.checkParemeterUpdate();
+    leftPIDController.checkParemeterUpdate();
   }
 
   public static Climb getInstance() {
     if (instance == null) {
-      if (Constants.getMode() == Mode.REAL && Constants.getRobot() != RobotType.ROBOT_DEFENSE) {
+      if (Constants.getMode() == Mode.REAL) {
```
Comment (jheidegger) [inline src/main/java/team3176/robot/subsystems/superstructure/climb/Climb.java:161]: re add this check
```suggestion
      if (Constants.getMode() == Mode.REAL && Constants.getRobot() != RobotType.ROBOT_DEFENSE) {
```

**Review by jheidegger — CHANGES_REQUESTED** [review]
I think the following might help clear things up even more

**Comment by char3176** [comment]
Before merging into integration branch let's see this on 2 motors (1 for each climb) simultaneously or at least in sim.

**Comment by jheidegger** [comment]
Andrew did you need to merge that in? Now we get all the merge conflicts we were trying to avoid

**Comment by AAgraviador3176** [comment]
@jheidegger If I'm being honest, I'm not entirely sure quite yet. We had too many branches for climb roll. They had different parts of the code on them. ClimbRoll2 and Z weren't working separately. Worst comes to worst, we revert to an old branch. I know it's not the perfect time to do this, but this was unfortunately, the solution that came to mind.

**Comment by jheidegger** [comment]
Gotcha gotcha yep we can cross the bridge when we come to it. Focus on getting it to work and I'll help make sure it can actually get merged in!

**Comment by char3176** [comment]
Resolved conflicts.  Merging into integration branch.

**Pull request merged**